src/lib/apollo/initApolloBrowser.js

```diff
@@ -28,6 +29,18 @@ const errorLink = onError(({ graphQLErrors, networkError }) => {
   }
 });
 
+const authLink = setContext((__, { headers }) => {
+  // get the authentication token from local storage if it exists
+  const token = localStorage.getItem("kc-token");
+  // return the headers to the context so httpLink can read them
+  return {
+    headers: {
+      ...headers,
+      Authorization: token ? `Bearer ${token}` : ""
+    }
+  };
+});
```

Comment:
It might just be that I'm missing something, but in order for SSR pages to be rendered with correct data based on auth, don't you have to use cookies the same way we're doing the `meteor-login-token` auth header? You could just add them to the existing `httpLink` functions in initApolloBrowser and initApolloServer, using `options.authToken` (or maybe use `options.token` and rename the existing to `options.meteorToken`). The handling for the server side session cookie is in the `if (!process.browser)` block in `withData.js`

Reply:
@aldeed So cookies actually didn't have anything to do with the auth token being available on the devserver side.  Every Next.js request originates from the client side and the headers are always being set there, so even in the case of an initial SSR'd request the headers are always present.

The cookies weren't actually doing anything for actual API auth before.  For example, [this token](https://github.com/reactioncommerce/reaction-next-starterkit/blob/master/src/lib/apollo/withApolloClient.js#L52-L61) returned `undefined` 100% of the time regardless of the presence of cookies.

Anyway, I completely rewrote the Apollo configs to be isomorphic and to handle all auth tokens from localstorage (which Keycloak also does) and then pass them up as headers (which the devserver currently expects).  I think localstorage/headers are the more flexible options for several reasons, but the big one is having to whitelist domains in the devserver for any app that wants to send up auth credentials in cookies.  It gets pretty tedious pretty quickly.  Just throwing an auth token in the headers removes that restriction and is a lot more in line with most standalone API's.  Requiring cookies for auth makes the assumption the client will always be a browser.  Since that could easily not be the case at times, I thought headers just felt like the right decision.

But perhaps I'm missing something about why things were done that way in the first place.  Either way, it definitely wasn't doing anything to provide auth to the GraphQL backend, so this seemed like an improvement and a simplification.